packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -0,0 +1,469 @@
+      textDirection: TextDirection.ltr,
+      child: Scaffold(
+        appBar: widget.drawerBreakpoint.isActive(context)
+            ? widget.appBar ?? AppBar()
+            : null,
+        drawer: widget.drawerBreakpoint.isActive(context)
+            ? Drawer(
+                child: NavigationRail(
+                  extended: true,
+                  selectedIndex: widget.selectedIndex,
+                  destinations:
+                      widget.destinations!.map(_toRailDestination).toList(),
+                  onDestinationSelected: widget.onSelectedIndexChange,
+                ),
+              )
+            : null,
+        body: AdaptiveLayout(
+          horizontalBody: widget.horizontalBody,
+          bodyRatio: widget.bodyRatio,
+          internalAnimations: widget.internalAnimations,
+          primaryNavigation: widget.destinations != null &&
+                  widget.selectedIndex != null
+              ? SlotLayout(
+                  config: <Breakpoint, SlotLayoutConfig>{
+                    widget.mediumBreakpoint: SlotLayoutConfig(
+                      key: const Key('primaryNavigation'),
+                      builder: (_) => SizedBox(
+                        width: 72,
+                        height: MediaQuery.of(context).size.height,
+                        child: NavigationRail(
+                          selectedIndex: widget.selectedIndex,
+                          destinations: widget.destinations!
+                              .map(_toRailDestination)
+                              .toList(),
+                          onDestinationSelected: widget.onSelectedIndexChange,
+                        ),
+                      ),
+                    ),
+                    widget.largeBreakpoint: SlotLayoutConfig(
+                      key: const Key('primaryNavigation1'),
+                      builder: (_) => SizedBox(
+                        width: 192,
+                        height: MediaQuery.of(context).size.height,
+                        child: NavigationRail(
+                          extended: true,
+                          selectedIndex: widget.selectedIndex,
+                          destinations: widget.destinations!
+                              .map(_toRailDestination)
+                              .toList(),
+                          onDestinationSelected: widget.onSelectedIndexChange,
+                        ),
+                      ),
+                    ),
+                  },
+                )
+              : null,
+          bottomNavigation: widget.destinations != null &&
+                  !widget.drawerBreakpoint.isActive(context)
+              ? SlotLayout(
+                  config: <Breakpoint, SlotLayoutConfig>{
+                    widget.smallBreakpoint: SlotLayoutConfig(
+                      key: const Key('bottomNavigation'),
+                      builder: (_) => BottomNavigationBar(
+                        unselectedItemColor: Colors.grey,
```

Comment:
These should definitely be configurable, or even better, just left unset so that people can use the default from the `BottomNavigationBarTheme`.